vespa-feed-client-cli/src/main/java/ai/vespa/feed/client/CliClient.java

```diff
@@ -60,22 +65,55 @@ private int run(String[] rawArgs) {
             try (InputStream in = createFeedInputStream(cliArgs);
                  FeedClient feedClient = createFeedClient(cliArgs);
                  JsonFeeder feeder = createJsonFeeder(feedClient, cliArgs)) {
+                CountDownLatch latch = new CountDownLatch(1);
+                AtomicReference<FeedException> fatal = new AtomicReference<>();
                 long startNanos = System.nanoTime();
-                feeder.feedMany(in).join();
+                feeder.feedMany(in, new ResultCallback() {
+                    @Override public void onNextResult(Result result, FeedException error) { handleResult(result, error, cliArgs); }
+                    @Override public void onError(FeedException error) { fatal.set(error); }
+                    @Override public void onComplete() { latch.countDown(); }
+                });
+                if (cliArgs.showProgress()) {
+                    new Thread(() -> {
```

Comment:
Free threads are ok since this is a CLI, otherwise this would be a potential resource leak. Make sure it's tagged as daemon.

Reply:
Will do.